converter/src/main/java/gov/cms/qpp/conversion/decode/QrdaDecoderEngine.java

```diff
@@ -165,6 +167,37 @@ private DecodeData decodeChildren(final Element element, final Node parentNode)
 		return decodeData;
 	}
 
+	/**
+	 * Reduces the {@code templateId} {@link Element}s so there are no duplicates.  All other {@link Element}s are left alone.
+	 *
+	 * @param childElements The elements to filter
+	 * @return A {@link List} of {@link Element}s that are filtered.
+	 */
+	private List<Element> getUniqueTemplateIdElements(final List<Element> childElements) {
+		Set<TemplateId> uniqueTemplates = EnumSet.noneOf(TemplateId.class);
+
+		return childElements.stream()
+			.filter(filterElement -> {
+				boolean isTemplateId = TEMPLATE_ID.equals(filterElement.getName());
+				TemplateId filterTemplateId = getTemplateId(filterElement);
+
+				boolean elementWillStay = true;
+
+				if (isTemplateId) {
+					if (getDecoder(filterTemplateId) == null) {
+						elementWillStay = false;
+					} else if (uniqueTemplates.contains(filterTemplateId)) {
```

Comment:
Can we add a test scenario where a Clinical Document template id w/o an extension precedes a valid CD w/ extension?